scripts/hid_configurator/configurator_core.py

```diff
@@ -0,0 +1,512 @@
+
+TYPE_FIELD_POS = 0
+GROUP_FIELD_POS = 6
+EVENT_GROUP_SETUP = 0x1
+EVENT_GROUP_DFU = 0x2
+
+MOD_FIELD_POS = 3
+SETUP_MODULE_SENSOR = 0x1
+SETUP_MODULE_LED = 0x1
+
+OPT_FIELD_POS = 0
+SENSOR_OPT_CPI = 0x0
+SENSOR_OPT_DOWNSHIFT_RUN = 0x1
+SENSOR_OPT_DOWNSHIFT_REST1 = 0x2
+SENSOR_OPT_DOWNSHIFT_REST2 = 0x3
+
+DFU_START = 0x0
+DFU_DATA = 0x1
+DFU_SYNC = 0x2
+DFU_REBOOT = 0x3
+DFU_IMGINFO = 0x4
+
+FLASH_PAGE_SIZE = 4096
+
+POLL_INTERVAL = 0.02
+POLL_RETRY_COUNT = 200
+
+DFU_SYNC_RETRIES = 3
+DFU_SYNC_INTERVAL = 1
+
+NORDIC_VID = 0x1915
+DEVICE_PID = {
+    'desktop_mouse_nrf52832'    : 0x52DA,
+    'desktop_mouse_nrf52810'    : 0x52DB,
+    'gaming_mouse'              : 0x52DE,
+    'keyboard'                  : 0x52DD,
+    'dongle'                    : 0x52DC,
+}
+
+
+class ConfigStatus(IntEnum):
+    SUCCESS            = 0
+    PENDING            = 1
+    FETCH              = 2
+    TIMEOUT            = 3
+    REJECT             = 4
+    WRITE_ERROR        = 5
+    DISCONNECTED_ERROR = 6
+    FAULT              = 99
+
+
+class Response(object):
+    def __init__(self, recipient, event_id, status, data):
+        self.recipient = recipient
+        self.event_id  = event_id
+        self.status    = ConfigStatus(status)
+        self.data      = data
+
+    def __repr__(self):
+        base_str = ('Response:\n'
+                    '\trecipient 0x{:04x}\n'
+                    '\tevent_id  0x{:02x}\n'
+                    '\tstatus    {}\n').format(self.recipient,
+                                               self.event_id,
+                                               str(self.status))
+        if self.data is None:
+            data_str = '\tno data'
+        else:
+            data_str = ('\tdata_len {}\n'
+                        '\tdata {}\n').format(len(self.data), self.data)
+
+        return base_str + data_str
+
+    @staticmethod
+    def parse_response(response_raw):
+        data_field_len = len(response_raw) - struct.calcsize('<BHBBB')
+
+        if data_field_len < 0:
+            logging.error('Response too short')
+            return None
+
+        # Report ID is not included in the feature report from device
+        fmt = '<BHBBB{}s'.format(data_field_len)
+
+        (report_id, rcpt, event_id, status, data_len, data) = struct.unpack(fmt, response_raw)
+
+        if report_id != REPORT_ID:
+            logging.error("Improper report ID")
+            return None
+
+        if data_len > len(data):
+            logging.error('Required data not present')
+            return None
+
+        if data_len == 0:
+            event_data = None
+        else:
+            event_data = data[:data_len]
+
+        return Response(rcpt, event_id, status, event_data)
+
+
+ConfigOption = collections.namedtuple('ConfigOption', 'range event_id help')
+SENSOR_OPTIONS = {
+    'downshift_run':    ConfigOption((10,   2550),   SENSOR_OPT_DOWNSHIFT_RUN,   'Time in milliseconds of switching from mode Run to Rest 1'),
```

Comment:
Maybe we could make descriptions shorter ?
Like : `Time of switching from mode Run to Rest 1 [ms]`
or even: `Switching time: Run -> Rest 1 [ms]`